components/eventing-controller/pkg/backend/jetstreamv2/jetstream.go

```diff
@@ -36,13 +36,14 @@ const (
 	jsConsumerAcKWait                  = 30 * time.Second
 )
 
-func NewJetStream(config env.NatsConfig, metricsCollector *backendmetrics.Collector, cleaner cleaner.Cleaner, logger *logger.Logger) *JetStream {
+func NewJetStream(config env.NatsConfig, metricsCollector *backendmetrics.Collector, cleaner cleaner.Cleaner, logger *logger.Logger, subsConfig env.DefaultSubscriptionConfig) *JetStream {
```

Comment:
(nit)
two points:
1) can we merge somehow the both of the configs being passed to the constructor ?
2) Usually the function arguments are listed by their priority, IMO logger is less priority than subConfig.

Reply:
Created a followup ticket: https://github.com/kyma-project/kyma/issues/15880